### PR TITLE
Unify, document and fix functional tests internal updating process

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Description

This fix the problem we had with the command supposed to update functional tests. It uniformize it between legacy and current functional tests, document it for legacy tests and fix it for current functional tests.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |
| ✓  | :hammer: Refactoring  |
| ✓  | :scroll: Docs |

### CODE REVIEW

File: doc/development_guide/contribute.rst [hunk 1]
```diff
@@ -162,12 +162,14 @@ During development, it's sometimes helpful to run all functional tests in your
 current environment in order to have faster feedback. Run from Pylint root directory with::
 
     python tests/test_functional.py
+    python tests/test_func.py # For legacy functional tests
```
Comment (PCManticore) [inline doc/development_guide/contribute.rst:165]: I think we should skip these. The intention is to remove them after they are ported to the new functional test framework. This hasn't been documented in a task, but it's something we should do to get rid of them.

File: tests/test_func.py [hunk 2]
```diff
@@ -158,11 +162,12 @@ def __test_functionality(module_file, messages_file, dependencies):
...
-        sys.argv.remove("-u")
-
+    if UPDATE_OPTION in sys.argv:
+        UPDATE_FILE.touch()
+        sys.argv.remove(UPDATE_OPTION)
     if len(sys.argv) > 1:
         FILTER_RGX = sys.argv[1]
         del sys.argv[1]
     pytest.main(sys.argv)
+    if UPDATE_FILE.exists():
```
Comment (PCManticore) [inline tests/test_func.py:172]: Is the intention here to delete it only after the tests complete successfully or in all the cases, including test failures? Worth putting in a `try-finally` if it's the latter.

File: doc/development_guide/contribute.rst (hunk 1 above)
Comment (Pierre-Sassoulas) [reply doc/development_guide/contribute.rst:165]: Good point, I'll remove that from the doc. The error message when a legacy tests fail tells dev what to do anyway.

File: tests/test_func.py (hunk 2 above)
Comment (Pierre-Sassoulas) [reply tests/test_func.py]: Fixed in https://github.com/PyCQA/pylint/pull/4004/commits/98dc5caeb3c737b9c1340cfe212de63986ac908c

**Review by PCManticore — APPROVED** [review]

**Pull request merged**